### PR TITLE
Refine Search interface

### PR DESCRIPTION
**Pull request by matrixji**
Signed-off-by: Ji Bin <matrixji@live.com>

### CODE REVIEW

File: src/include/types/CollectionSchema.h [hunk 1]
```diff
@@ -107,6 +107,20 @@ class CollectionSchema {
...
     }
 
+    /**
+     * @brief Return Anns field name, if not exist return ""
+     */
+    std::string
+    AnnsFieldName() const {
+        for (const auto& field : fields_) {
+            auto data_type = field.FieldDataType();
+            if (data_type == DataType::BINARY_VECTOR || data_type == DataType::FLOAT_VECTOR) {
```
Comment (yhmo) [inline src/include/types/CollectionSchema.h:117]: Currently, each collection only supports one vector field, this function is no problem. But in later version(maybe 2.1), each collection can manage multiple vector fields. This function could be a potential bug in future.

File: src/impl/MilvusClientImpl.cpp [hunk 2]
```diff
@@ -472,7 +472,17 @@ MilvusClientImpl::Delete(const std::string& collection_name, const std::string&
 
 Status
 MilvusClientImpl::Search(const SearchArguments& arguments, SearchResults& results) {
-    auto pre = [&arguments]() {
+    std::string anns_field;
+    auto validate = [this, &arguments, &anns_field]() {
+        CollectionDesc collection_desc;
+        auto status = DescribeCollection(arguments.CollectionName(), collection_desc);
+        if (status.IsOk()) {
+            anns_field = collection_desc.Schema().AnnsFieldName();
```
Comment (yhmo) [inline src/impl/MilvusClientImpl.cpp:480]: I suggest add a method "SetAnnsFieldName()" to the SearchArguments class, or change the AddTargetVectors() to be:
`Status AddTargetVector(const std::string& field_name, const FloatVecFieldData::ElementT& vector)`

Then create FieldData with the name, if the FieldData already exists, just return an error. 
Now we only allow search one vector field, in someday maybe we can allow two vectors fields to search together with  different weight.

Comment (matrixji) [reply src/impl/MilvusClientImpl.cpp:480]: Seems `Status AddTargetVector(const std::string& field_name, const FloatVecFieldData::ElementT& vector)` is more reasonable right now.

File: src/include/types/CollectionSchema.h (hunk 1 above)
Comment (matrixji) [reply src/include/types/CollectionSchema.h:117]: I'd like refactor to `std::unorder_set<std::string> AnnsFieldNames()` and use it for validation.

Comment (yhmo) [reply src/include/types/CollectionSchema.h:117]: Ok, return `std::unordered_set<std::string>` is reasonable.

**Pull request merged**